### PR TITLE
[12.0][ADD] stock_location_limit_product

### PR DESCRIPTION
**Pull request by nikitavaghela**
Stock Location Limit Product
======================
Added stock_location_limit_product as mention #756

### CODE REVIEW

File: stock_location_limit_product/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    'name': 'Stock Location Limit Product',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA),',
+    'website': "https://github.com/OCA/stock-logistics-warehouse",
+    'summary': """Add a limit by product quantity on a stock location.
+                This limit can later be used to track the capacity
+                available of the location.""",
+    'category': 'Stock Management',
```
Comment (rousseldenis) [inline stock_location_limit_product/__manifest__.py]: Set this to 'Warehouse' as it won't appear in good category in OCA shop

File: stock_location_limit_product/models/stock_location_limit.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+from odoo import api, fields, models
+
+
+class StockLocationLimit(models.Model):
+    _name = 'stock.location.limit'
+    _description = 'Stock Location Limit Product'
+    _rec_name = 'product_id'
+
+    _sql_constraints = [
```
Comment (rousseldenis) [inline stock_location_limit_product/models/stock_location_limit.py]: You have to take into account the 'active' field of the product.

To do such constrain, see this : https://github.com/OCA/partner-contact/blob/10.0/partner_bank_sort_code/models/res_bank.py#L16

File: stock_location_limit_product/models/stock_location_limit.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        ('product_uniq', 'unique(product_id,location_id)',
+            'Product and Location must be unique!'),
+    ]
+
+    product_id = fields.Many2one(
+        'product.product',
+        string='Product',
+    )
+    qty = fields.Integer('Maximum Quantity')
```
Comment (rousseldenis) [inline stock_location_limit_product/models/stock_location_limit.py]: You need to use Float with decimal precision or your module would have usage limitations.

File: stock_location_limit_product/models/stock_location_limit.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    uom_id = fields.Many2one(
+        'uom.uom',
+        string='UoM',
+        default=lambda self: self.product_id.uom_id.id
```
Comment (rousseldenis) [inline stock_location_limit_product/models/stock_location_limit.py]: Set this as related on product_id.uom_id. Easier + onchange removal

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

As I understand, this is a base module for further ones ?

File: stock_location_limit_product/models/stock_location_limit.py (hunk 2 above)
Comment (nikitavaghela) [reply stock_location_limit_product/models/stock_location_limit.py]: hello @rousseldenis, can you suggest to me the reference for  EXCLUDE constraint for the relational table? I checked this reference but as per my knowledge, we can only use the current table's column in the constraint.

Comment (rousseldenis) [reply stock_location_limit_product/models/stock_location_limit.py]: Mhh, maybe my remark has no sense as if a product is not active anymore, there is no reason to create stock location limit.

Comment (nikitavaghela) [reply stock_location_limit_product/models/stock_location_limit.py]: yes  :) so can I add simple unique constrain as before?

File: stock_location_limit_product/models/stock_location_limit.py [hunk 5]
```diff
@@ -19,23 +19,13 @@ class StockLocationLimit(models.Model):
         'product.product',
         string='Product',
     )
-    qty = fields.Integer('Maximum Quantity')
+    qty = fields.Float('Maximum Quantity')
```
Comment (rousseldenis) [inline stock_location_limit_product/models/stock_location_limit.py]: Don't forget decimal precision

**Review by rousseldenis — APPROVED** [review]
Code review.

Could you squash your commits ?

**Comment by nikitavaghela** [comment]
> Code review.
> 
> Could you squash your commits ?

**Comment by nikitavaghela** [comment]
> Code review.
> Could you squash your commits ?

I tried to squash but there are some other commits also between our module commit. so it's not sqashed.

**Comment by rousseldenis** [comment]
What did you do to squash (commands) ?

**Comment by nikitavaghela** [comment]
> What did you do to squash (commands) ?

git rebase -i HEAD~(no of commit to squash)

**Comment by nikitavaghela** [comment]
Hello @max3903, @dreispt,
Can you please review this?

File: stock_location_limit_product/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                available of the location.""",
+    'category': 'Warehouse',
+    'depends': [
+        'stock',
+    ],
+    'data': [
+        'security/ir.model.access.csv',
+        'views/stock_location_view.xml',
+    ],
+    'installable': True,
```
Comment (max3903) [inline stock_location_limit_product/__manifest__.py:25]:
```suggestion
    'installable': True,
    'development_status': 'Stable',
    'maintainers': ['max3903'],
```

**Comment by max3903** [comment]
@nikitavaghela Please add a test that adds a limit record to the stock location.

**Comment by max3903** [comment]
@nikitavaghela Please review and merge https://github.com/nikitavaghela/stock-logistics-warehouse/pull/1

**Review by max3903 — APPROVED** [review]

**Pull request merged**